src/cpptools.ts

```diff
@@ -424,14 +427,19 @@ export class CppConfigurationProvider implements cpptools.CustomConfigurationPro
      * @param fileGroup The file group from the code model to create config data for
      * @param opts Index update options
      */
-    private buildConfigurationData(fileGroup: codeModel.CodeModelFileGroup, opts: codeModel.CodeModelParams, target: TargetDefaults, sysroot: string): cpptools.SourceFileConfiguration {
+    private buildConfigurationData(fileGroup: CodeModelFileGroup, opts: CodeModelParams, target: TargetDefaults, sysroot?: string): cpptools.SourceFileConfiguration {
+        // For CppTools V6 and above, build the compilerFragments data, otherwise build compilerArgs data
+        const useFragments: boolean = this.cpptoolsVersion >= cpptools.Version.v6;
```

Comment:
Let's just check if fragments exist on the codemodel instead of making all the changes to the server codemodel.  If the fragments don't exist, then it must be a server codemodel object and we want to use flags instead.